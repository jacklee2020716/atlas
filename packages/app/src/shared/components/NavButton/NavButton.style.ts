import styled from '@emotion/styled'
import { colors } from '../../theme'
import Button from '../Button'
export type NavButtonStyleProps = {
  variant: 'primary' | 'secondary'
}

export const StyledButton = styled(Button)`
  color: ${(props) => (props.variant === 'primary' ? colors.white : colors.gray[600])};
  background-color: ${(props) => (props.variant === 'primary' ? colors.blue[500] : 'transparent')};
  border: unset;
  width: 48px;
  height: 48px;
  &:hover {
    color: ${colors.white};
    background-color: ${(props) => (props.variant === 'primary' ? colors.blue[700] : 'transparent')};
  }
  &:active {
    background-color: ${(props) => (props.variant === 'primary' ? colors.blue[900] : 'transparent')};
  }
`
