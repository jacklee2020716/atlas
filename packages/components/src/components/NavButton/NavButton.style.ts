import { typography, colors } from "../../theme"
import { StyleFn, makeStyles } from "../../utils"

export type NavButtonStyleProps = {
	type?: "primary" | "secondary"
}

const baseStyles: StyleFn = () => ({
	border: 0,
	color: colors.white,
	textAlign: "center",
	display: "inline-block",
	cursor: "default",
	fontFamily: typography.fonts.base,
	fontWeight: typography.weights.medium,
	fontSize: typography.sizes.subtitle1,
	margin: "1px",
	padding: 0,
	lineHeight: "50px",
	"&:hover": {
		borderColor: colors.blue[700]
	},
	"&:active": {
		borderColor: colors.blue[900]
	},
	"&::selection": {
		background: "transparent"
	}
})

const colorFromType: StyleFn = (styles, { type = "primary" }) => ({
	...styles,
	backgroundColor: type === "primary" ? colors.blue[700] : colors.black,
	"&:hover": {
		backgroundColor: type === "primary" ? colors.blue[700] : colors.black,
		color: type === "primary" ? colors.white : colors.blue[300]
	},
	"&:active": {
		backgroundColor: type === "primary" ? colors.blue[900] : colors.black,
		color: type === "primary" ? colors.white : colors.blue[700]
	}
})

export const useCSS = (props: NavButtonStyleProps) => makeStyles([baseStyles, colorFromType])(props)
