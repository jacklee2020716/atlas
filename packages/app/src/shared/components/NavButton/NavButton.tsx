import React from 'react'
import { SerializedStyles } from '@emotion/core'
import { NavButtonStyleProps, useCSS } from './NavButton.style'
import { ChevronLeftIcon, ChevronRightIcon } from '../../icons'

type NavButtonProps = {
  direction: 'right' | 'left'
  outerCss: SerializedStyles | SerializedStyles[] | (SerializedStyles | undefined) | (SerializedStyles | undefined)[]
  onClick: (e: React.MouseEvent<HTMLButtonElement>) => void
} & NavButtonStyleProps

export default function NavButton({ direction = 'right', onClick, outerCss, ...styleProps }: Partial<NavButtonProps>) {
  const styles = useCSS(styleProps)
  return (
    <button css={[styles, outerCss]} onClick={onClick}>
      {direction === 'right' ? <ChevronRightIcon /> : <ChevronLeftIcon />}
    </button>
  )
}
