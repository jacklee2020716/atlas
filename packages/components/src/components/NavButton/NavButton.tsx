import React from "react";
import { SerializedStyles } from "@emotion/core";
import { useCSS, NavButtonStyleProps } from "./NavButton.style";
import ChevronRightIcon from "../../../assets/chevron-right-big.svg";
import ChevronLeftIcon from "../../../assets/chevron-left-big.svg";

type NavButtonProps = {
	direction: "right" | "left";
	outerCss: SerializedStyles | SerializedStyles[] | (SerializedStyles | undefined) | (SerializedStyles | undefined)[];
	onClick: (e: React.MouseEvent<HTMLButtonElement>) => void;
} & NavButtonStyleProps;

export default function NavButton({ direction = "right", onClick, outerCss, ...styleProps }: Partial<NavButtonProps>) {
	let styles = useCSS(styleProps);
	return (
		<button css={[styles, outerCss]} onClick={onClick}>
			{direction === "right" ? <ChevronRightIcon /> : <ChevronLeftIcon />}
		</button>
	);
}
