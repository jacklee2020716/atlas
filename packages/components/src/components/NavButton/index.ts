import { memo } from "react"
import NavButton from "./NavButton"
export default memo(NavButton)
