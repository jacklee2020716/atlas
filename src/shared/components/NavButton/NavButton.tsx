import React from 'react'
import { SerializedStyles } from '@emotion/core'
import { NavButtonStyleProps, StyledButton } from './NavButton.style'
import Icon from '../Icon'

type NavButtonProps = {
  direction: 'right' | 'left'
  outerCss: SerializedStyles | (SerializedStyles | undefined)[]
  onClick: (e: React.MouseEvent<HTMLButtonElement>) => void
} & NavButtonStyleProps

const NavButton: React.FC<Partial<NavButtonProps>> = ({
  direction = 'right',
  onClick,
  outerCss,
  variant = 'primary',
}) => {
  return (
    <StyledButton css={outerCss} onClick={onClick} variant={variant}>
      <Icon name={direction === 'right' ? 'chevron-right' : 'chevron-left'} />
    </StyledButton>
  )
}

export default NavButton
