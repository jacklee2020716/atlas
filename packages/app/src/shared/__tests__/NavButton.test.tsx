import React from 'react'
import { mount } from 'enzyme'
import NavButton from '@/shared/components/NavButton'
describe('NavButton component', () => {
  it('Should render NavButton correctly', () => {
    expect(mount(<NavButton />)).toMatchSnapshot()
  })
})
