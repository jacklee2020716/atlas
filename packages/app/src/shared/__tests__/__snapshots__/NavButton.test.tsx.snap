// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavButton component Should render NavButton correctly 1`] = `ReactWrapper {}`;
